### PR TITLE
ref: fix type ignores in organization_events_trends_v2

### PR DESCRIPTION
**Pull request by asottile-sentry**

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_trends_v2.py [hunk 1]
```diff
@@ -129,15 +129,9 @@ def get_timeseries(top_events, _, rollup, zerofill_results):
...
+            timeseries_columns = selected_columns[:]
             timeseries_columns.append(trend_function)
 
-            # When all projects or my projects options selected,
-            # keep only projects that top events belong to to reduce query cardinality
-            used_project_ids = list({event["project"] for event in data})
-
-            request.GET.projectSlugs = used_project_ids  # type: ignore[attr-defined]
-
             # Get new params with pruned projects
```
Comment (asottile-sentry) [inline src/sentry/api/endpoints/organization_events_trends_v2.py:135]: this appears to be completely bogus -- and the type checker was helpfully pointing it out and was ignored :(

Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_trends_v2.py:135]: I agree that this is poorly written but I think deleting this has the effect of scanning additional rows in the following clickhouse query.

Do you mind reverting this part and prune `pruned_params` below to contain just the projects in `data`? I can also take this part from you and push a fix today if you want.

Comment (asottile-sentry) [reply src/sentry/api/endpoints/organization_events_trends_v2.py:135]: the `projectSlugs` token does not appear in the codebase at all -- I don't think this is doing what is intended -- and it's certainly not demonstrated in a test

could you point me at why you think this code is doing something?

Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_trends_v2.py:135]: I believe it's done here https://github.com/getsentry/sentry/blob/master/src/sentry/api/bases/organization.py#L339

Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_trends_v2.py:135]: I fixed this in #67701, feel free to either merge that one or pull the changes in this PR

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**